zeth-contracts/contracts/BaseMixer.sol

```diff
@@ -111,81 +111,44 @@ contract BaseMixer is MerkleTreeMiMC7, ERC223ReceivingContract {
     // This function processes the primary inputs to append and check the root and nullifiers in the primary inputs (instance)
     // and modifies the state of the mixer contract accordingly
     // (ie: Appends the commitments to the tree, appends the nullifiers to the list and so on)
-    function assemble_root_and_nullifiers_and_append_to_state(uint[] memory primary_inputs) internal {
-        // 1. We re-assemble the full root digest from the 2 field elements it was packed into
-        uint256[] memory digest_inputs = new uint[](2);
-        digest_inputs[0] = primary_inputs[0];
+    function assemble_and_check_root_nullifiers_and_vk_and_append_to_state(
```

Comment:
Let's rename this function, the name becomes very long and unclear.
This function checks:
1. The mk_root represent a past (or the current) merkle tree of note commitments
2. The nullifiers have not be declared already (no double spend)
3. The h_sig is valid

So let's rename the function accordingly. Something like `check_mkroot_nullifiers_hsig(...)` and please make sure the comments documenting the function are up to date

Reply:
Fixed

Reply:
Done